doc/adr/0011-tsp-analysis-api.md

```diff
@@ -8,124 +8,124 @@ Proposed
-    GET /analysis/types/{typeId}
+    GET /config/
+        returns a list of configuration source types: typeId, name, description, expected input parameter list ("path" for file path)
+    GET /config/types
+        returns a map typeId -> list of configuration instances configured on the server for a given configuration source type: configId, name, description, parameters
+    GET /config/types/{typeId}
         returns a list of analysis descriptors for given typeId
-    POST /analysis/types/{typeId}
-        Upload an analysis definition for given typeId
-        Returns a new analysis descriptor with unique analysisID
-    PUT /analysis/types/{typeId}/{analysisId}
-        Update an analysis
-    GET /analysis/types/{typeId}/{analysisId}
-        Returns the analysis descriptor for given analysisId
-    DELETE /analysis/types/{typeId}/{analysisId}
-        Delete an analysis
-
-#### Analysis type descriptor
-
-The analysis type descriptor describes the type of external analysis. Different types have different syntax and hence back-end implementation. This descriptor is used to distinquish the different external analysis. The trace server implementation will provide the list of type descriptors on client requests.
+    POST /config/types/{typeId}
+        Upload an configuration definition for given typeId
+        Returns a new configuration descriptor with unique configuration ID
+    PUT /config/types/{typeId}/{configId}
+        Update a configuration
+    GET /config/types/{typeId}/{configId}
+        Returns the configuration descriptor for given configId and typeId
+    DELETE /config/types/{typeId}/{configId}
+        Delete a configuration instance
+
+#### Configuration source type descriptor
+
+The configuration source type descriptor describes the type of external configuration. Different types have different syntax and hence back-end implementation. This descriptor is used to distinquish the different external configurations. The trace server implementation will provide the list of type descriptors on client requests.
 ```javascript
-AnalysisTypeDescriptor {
+ConfigurationSourceType {
     name: string,
     description: string,
     id: string,
     scope: string,
-    inputParameter: string[]
+    queryParameterKeys: string[]
 }
 ```
 
 Where:
 
-- `name`: Name of the analysis type. Can be shown to the end-user.
-- `description`: The description of the analysis type. Can be shown to the end-user.
-- `id`: Unique id of the analysis type. Used in the application to distinquish analysis types
-- `scope:` `experiment` for analysis per experiment or `global` for all experiments
+- `name`: Name of the configuration source type. Can be shown to the end-user.
+- `description`: The description of the configuration source type. Can be shown to the end-user.
+- `id`: Unique id of the configuration source type. Used in the application to distinquish configuration source types
```

Comment:
id -> typeId

Reply:
The data structure already indicates that's about types. So we don't need to change it to typeId.